From c9d202e6e5a6390960de012cbccb01bf90a98963 Mon Sep 17 00:00:00 2001
From: simon_fan <simon_fan@edge-core.com>
Date: Sat, 23 Sep 2023 03:35:35 +0000
Subject: [PATCH] Add com-e ec driver

---
 drivers/hwmon/Kconfig  | 5 +++++
 drivers/hwmon/Makefile | 1 +
 2 files changed, 6 insertions(+)

diff --git a/drivers/hwmon/Kconfig b/drivers/hwmon/Kconfig
index d05656d2332b..e61dbfe68e72 100644
--- a/drivers/hwmon/Kconfig
+++ b/drivers/hwmon/Kconfig
@@ -1966,6 +1966,11 @@ config SENSORS_EC_SYSTEM_FPGA
 	help
 	  If you say yes here you get support for the EC System FPGA.
 
+config SENSORS_COM_E_DRV
+	tristate "Edge-core COMe EC"
+        help
+	  If you say yes here you get hardware monitoring support Edge-core COMe EC.
+
 if ACPI
 
 comment "ACPI drivers"
diff --git a/drivers/hwmon/Makefile b/drivers/hwmon/Makefile
index a305df93d6d5..d12decff12c3 100644
--- a/drivers/hwmon/Makefile
+++ b/drivers/hwmon/Makefile
@@ -182,6 +182,7 @@ obj-$(CONFIG_SENSORS_XGENE)	+= xgene-hwmon.o
 obj-$(CONFIG_I2C_MUX_EC_FPGA_MUX) += ec_fpga_mux.o
 obj-$(CONFIG_SENSORS_EC_FAN_CPLD) += ec_fan_cpld.o
 obj-$(CONFIG_SENSORS_EC_SYSTEM_FPGA) += ec_system_fpga.o
+obj-$(CONFIG_SENSORS_COM_E_DRV) += ec_com_e_drv.o
 
 obj-$(CONFIG_SENSORS_OCC)	+= occ/
 obj-$(CONFIG_PMBUS)		+= pmbus/
-- 
2.17.1

