From 2986fd7333de83b4ed6f986d38015affcae646a6 Mon Sep 17 00:00:00 2001
From: simon_fan <simon_fan@edge-core.com>
Date: Thu, 8 Oct 2020 09:05:40 +0800
Subject: [PATCH] Add ast2600-ec-as9737-32db.dts device tree

---
 arch/arm/dts/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index e4dae29379..a642a3de39 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -683,6 +683,7 @@ dtb-$(CONFIG_ARCH_ASPEED) += \
 	ast2600-fpga.dtb \
 	ast2600-rainier.dtb \
 	ast2600-slt.dtb \
+	ast2600-ec-as9737-32db.dtb \
 	ast2600-tacoma.dtb
 
 dtb-$(CONFIG_ARCH_STI) += stih410-b2260.dtb
-- 
2.17.1

